stellarphot/core.py

```diff
@@ -62,3 +70,305 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
+    fwhm_x                u.pix
+    fwhm_y                u.pix
+    width                 u.pix
+    aperture              u.pix
+    annulus_inner         u.pix
+    annulus_outer         u.pix
+    aperture_sum          consistent count units
+    annulus_sum           consistent count units
+    sky_per_pix_avg       consistent count units
+    sky_per_pix_med       consistent count units
+    sky_per_pix_std       consistent count units
+    aperture_net_cnts     consistent count units
+    noise                 consistent count units
+    exposure              u.s
+    date-obs              astropy.time.Time with scale='utc'
+    airmass               None
+    passband              None
+    file                  None
+
+    In addition to these required columns, the following columns are computed based
+    on the input data during creation.
+
+    aperture_area
+    annulus_area
+    snr
+    bjd
+    night
+    mag_inst
+    mag_error
+
+    If these computed columns already exist in `data` class the class
+    will throw an error a ValueError UNLESS`ignore_computed=True`
+    is passed to the initializer, in which case the columns will be
+    retained and not replaced with the computed values.
+
+    Attributes
+    ----------
+    camera: `stellarphot.Camera`
+        A description of the CCD used to perform the photometry.
+
+    data: `astropy.table.Table`
+        A table containing all the instrumental aperture photometry results.
+
+    observatory: `astropy.coordinates.EarthLocation`
+        The location of the observatory.
+    """
+
+    # Define columns in the photo_table and provide information about their type, and units.
+    phot_descript = {
+        'star_id' : None,
+        'ra' : u.deg,
+        'dec' : u.deg,
+        'xcenter' : u.pix,
+        'ycenter' : u.pix,
+        'fwhm_x' : u.pix,
+        'fwhm_y' : u.pix,
+        'width' : u.pix,
+        'aperture' : u.pix,
+        'annulus_inner' : u.pix,
+        'annulus_outer' : u.pix,
+        'aperture_sum' : None,
+        'annulus_sum' : None,
+        'sky_per_pix_avg' : None,
+        'sky_per_pix_med' : None,
+        'sky_per_pix_std' : None,
+        'aperture_net_cnts' : None,
+        'noise' : None,
+        'exposure' : u.s,
+        'date-obs' : None,
+        'airmass' : None,
+        'passband' : None,
+        'file' : None
+    }
+
+    def __init__(self, observatory, camera, data,  passband_map=None, retain_user_computed=False):
+        # Set metavariables
+        self.observatory = observatory
+        self.camera = camera
+        self.passband_map = passband_map
+
+        # Check the time column is correct format.
+        # Correct dtype and class
+        if (data['date-obs'].dtype != object) or (data['date-obs'][0].__class__ != Time):
+            raise ValueError(f"data['date-obs'] is not column of astropy.time.Time objects.")
+        # Correct scale is set for first element
+        if (data['date-obs'][0].scale != 'utc'):
```

Comment:
Can this be consolidated with the previous `if` like this:

```python
try:
    if (data['date-obs'][0].scale != 'utc'):
        raise ValueError(f"data['date-obs'] astropy.time.Time must have scale='utc', not \'{data['date-obs'][0].scale}\'.")
# Happens if the first item does not have a "scale"
except AttributeError:
    raise ValueError(f"data['date-obs'] is not column of astropy.time.Time objects.")

Reply:
OK, fixed.